res/controllers/common-controller-scripts.js

```diff
@@ -418,29 +419,49 @@ bpm.tap = []   // Tap sample values
    Output:  -
    -------- ------------------------------------------------------ */
 bpm.tapButton = function(deck) {
-    var now = new Date() / 1000   // Current time in seconds
-    var tapDelta = now - bpm.tapTime
-    bpm.tapTime = now
-    if (tapDelta > 2.0) { // reset if longer than two seconds between taps
-        bpm.tap = []
-        return
-    }
-    bpm.tap.push(60 / tapDelta)
-    if (bpm.tap.length > 8) bpm.tap.shift()  // Keep the last 8 samples for averaging
-    var sum = 0
-    for (i = 0; i < bpm.tap.length; i++) {
-        sum += bpm.tap[i]
+    var now = new Date() / 1000; // Current time in seconds
+    var tapDelta = now - bpm.tapTime;
+    bpm.tapTime = now;
+
+    // assign tapDelta in cases where the button has not been pressed previously
+    if (bpm.tap.length < 1) {
+        bpm.previousTapDelta = tapDelta;
+    }
+    // reset if longer than two seconds between taps
+    if (tapDelta > 2.0) {
+        bpm.tap = [];
+        return;
+    }
+    // reject occurences of accidental double or missed taps
+    // a tap is considered missed when the delta of this press is 80% longer than the previous one
+    // and a tap is considered double when the delta is shorter than 40% of the previous one.
+    // these numbers are just guesses that produced good results in practice
+    if ((tapDelta > bpm.previousTapDelta*1.8)||(tapDelta < bpm.previousTapDelta*0.4)) {
+        return;
+    }
+    bpm.previousTapDelta = tapDelta;
+    bpm.tap.push(60 / tapDelta);
+    // Keep the last 8 samples for averaging
+    if (bpm.tap.length > 8) bpm.tap.shift();
+    var sum = 0;
+    for (i=0; i<bpm.tap.length; i++) {
+        sum += bpm.tap[i];
     }
     var average = sum / bpm.tap.length
 
-    var fRateScale = average / engine.getValue("[Channel" + deck + "]", "bpm")
+    // "bpm" was changed in 1.10 to reflect the *adjusted* bpm, but I presume it
+    // was supposed to return the tracks bpm (which it did before the change).
+    // "file_bpm" is supposed to return the set BPM of the loaded track of the
+    // channel.
+    var fRateScale = average/engine.getValue("[Channel" + deck + "]", "file_bpm");
```

Comment:
Since you use this multiple times, can you save it into a variable? I think this would improve readability.
```js
var group = "[Channel" + deck + "]";
```